File: benchmark_phase_c/bench_multi.cs
using System;

interface IFlyable
{
    int Fly();
}

class Animal
{
    public int sound;
    public Animal(int sound) { this.sound = sound; }
    public virtual int Speak() { return sound; }
}

class Bird : Animal, IFlyable
{
    public Bird(int sound) : base(sound) { }
    public int Fly() { return sound * 10; }
}

class Program
{
    static int BenchMulti(int n)
    {
        var b = new Bird(3);
        int total = 0;
        int i = 0;
        while (i < n)
        {
            total = total + ((b.Speak() + b.Fly()) % 1000);
            i = i + 1;
        }
        return total;
    }

    static void Main(string[] args)
    {
        int n = int.Parse(args[0]);
        Console.WriteLine(BenchMulti(n));
    }
}

File: benchmark_phase_c/bench_multi.py
import sys


class Flyable:
    def fly(self):
        pass


class Animal:
    def __init__(self, sound):
        self.sound = sound

    def speak(self):
        return self.sound


class Bird(Animal, Flyable):
    def __init__(self, sound):
        super().__init__(sound)

    def fly(self):
        return self.sound * 10


def bench_multi(n):
    b = Bird(3)
    total = 0
    i = 0
    while i < n:
        total = total + ((b.speak() + b.fly()) % 1000)
        i = i + 1
    return total


if __name__ == '__main__':
    n = int(sys.argv[1])
    print(bench_multi(n))
